C++ error: default-arguments are not allowed on this symbol

It seems that there is an [Error C2383](https://msdn.microsoft.com/en-us/library/y81x4ttb.aspx) in [SerializeAST.cc lines 596](https://github.com/apiaryio/drafter/blob/master/src/SerializeAST.cc#L596) and [600](https://github.com/apiaryio/drafter/blob/master/src/SerializeAST.cc#L600)

Do you have a quick workaround to allow me to build it?

**Complete output:**

```
C:\Projects\drafter>vcbuild.bat
creating config.gypi
creating config.mk
creating makefiles
Warning: Missing input files:
./build\ext\snowcrash\..\..\..\ext\snowcrash\src\Version.h
All OK.
Project files generated.
Building Snow Crash...
Build started 10/16/2015 10:16:41 AM.
     1>Project "C:\Projects\drafter\build\drafter.sln" on node 1 (default targets).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Release|Win32".
     1>Project "C:\Projects\drafter\build\drafter.sln" (1) is building "C:\Projects\drafter\build\test-libdrafter.vcxpr
       oj.metaproj" (3) on node 1 (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       libsos.vcxproj" (4) on node 1 (default targets).
     4>InitializeBuildStatus:
         Creating "Release\obj\libsos\libsos.unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libsos.vcxproj -> C:\Projects\drafter\build\Release\lib\libsos.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libsos\libsos.unsuccessfulbuild".
         Touching "Release\obj\libsos\libsos.lastbuildstate".
     4>Done Building Project "C:\Projects\drafter\build\libsos.vcxproj" (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       ext\snowcrash\libsnowcrash.vcxproj" (5) on node 1 (default targets).
     5>InitializeBuildStatus:
         Creating "Release\obj\libsnowcrash\libsnowcrash.unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libsnowcrash.vcxproj -> C:\Projects\drafter\build\Release\lib\libsnowcrash.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libsnowcrash\libsnowcrash.unsuccessfulbuild".
         Touching "Release\obj\libsnowcrash\libsnowcrash.lastbuildstate".
     5>Done Building Project "C:\Projects\drafter\build\ext\snowcrash\libsnowcrash.vcxproj" (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       ext\snowcrash\libmarkdownparser.vcxproj" (6) on node 1 (default targets).
     6>InitializeBuildStatus:
         Creating "Release\obj\libmarkdownparser\libmarkdownparser.unsuccessfulbuild" because "AlwaysCreate" was specif
         ied.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libmarkdownparser.vcxproj -> C:\Projects\drafter\build\Release\lib\libmarkdownparser.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libmarkdownparser\libmarkdownparser.unsuccessfulbuild".
         Touching "Release\obj\libmarkdownparser\libmarkdownparser.lastbuildstate".
     6>Done Building Project "C:\Projects\drafter\build\ext\snowcrash\libmarkdownparser.vcxproj" (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       ext\snowcrash\libsundown.vcxproj" (7) on node 1 (default targets).
     7>InitializeBuildStatus:
         Creating "Release\obj\libsundown\libsundown.unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libsundown.vcxproj -> C:\Projects\drafter\build\Release\lib\libsundown.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libsundown\libsundown.unsuccessfulbuild".
         Touching "Release\obj\libsundown\libsundown.lastbuildstate".
     7>Done Building Project "C:\Projects\drafter\build\ext\snowcrash\libsundown.vcxproj" (default targets).
     1>Project "C:\Projects\drafter\build\drafter.sln" (1) is building "C:\Projects\drafter\build\drafter.vcxproj.metap
       roj" (2) on node 1 (default targets).
     2>Project "C:\Projects\drafter\build\drafter.vcxproj.metaproj" (2) is building "C:\Projects\drafter\build\libdraft
       er.vcxproj" (8) on node 2 (default targets).
     8>InitializeBuildStatus:
         Creating "Release\obj\libdrafter\libdrafter.unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         C:\Program Files\Microsoft Visual Studio 11.0\VC\bin\CL.exe /c /I"..\ext\snowcrash\ext\markdown-parser\src" /I
         "..\ext\snowcrash\ext\markdown-parser\ext\sundown\src" /I..\ext\sos\src /I..\ext\snowcrash\src /Zi /nologo /W1
          /WX- /Od /Ob2 /Oi /Ot /Oy /GL /D BUILDING_SNOWCRASH=1 /D WIN32 /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO
         _DEPRECATE /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /GR /Fo"Release\obj\libdrafter\\" /Fd"R
         elease\obj\libdrafter\vc110.pdb" /Gd /TP /analyze- /errorReport:queue /MP ..\src\SerializeAST.cc
         SerializeAST.cc
     8>..\src\SerializeAST.cc(596): error C2383: 'Functor' : default-arguments are not allowed on this symbol [C:\Proje
       cts\drafter\build\libdrafter.vcxproj]
     8>..\src\SerializeAST.cc(600): error C2383: 'Functor' : default-arguments are not allowed on this symbol [C:\Proje
       cts\drafter\build\libdrafter.vcxproj]
     8>Done Building Project "C:\Projects\drafter\build\libdrafter.vcxproj" (default targets) -- FAILED.
     2>Done Building Project "C:\Projects\drafter\build\drafter.vcxproj.metaproj" (default targets) -- FAILED.
     3>Done Building Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (default targets) -- FAILED.
     1>Done Building Project "C:\Projects\drafter\build\drafter.sln" (default targets) -- FAILED.

Build FAILED.

Time Elapsed 00:00:03.12
```

Commit 0612f337dc1ee8427fd2f297d2708e3f4bdbe1da builds successfully, so quick workaround is not need. Thanks for you quick response!

**Complete output:**

```
C:\Projects\drafter>vcbuild.bat
creating config.gypi
creating config.mk
creating makefiles
Warning: Missing input files:
./build\ext\snowcrash\..\..\..\ext\snowcrash\src\Version.h
All OK.
Project files generated.
Building Snow Crash...
Build started 10/16/2015 10:26:26 AM.
     1>Project "C:\Projects\drafter\build\drafter.sln" on node 1 (default targets).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Release|Win32".
     1>Project "C:\Projects\drafter\build\drafter.sln" (1) is building "C:\Projects\drafter\build\test-libdrafter.vcxpr
       oj.metaproj" (3) on node 1 (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       libsos.vcxproj" (4) on node 1 (default targets).
     4>InitializeBuildStatus:
         Creating "Release\obj\libsos\libsos.unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libsos.vcxproj -> C:\Projects\drafter\build\Release\lib\libsos.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libsos\libsos.unsuccessfulbuild".
         Touching "Release\obj\libsos\libsos.lastbuildstate".
     4>Done Building Project "C:\Projects\drafter\build\libsos.vcxproj" (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       ext\snowcrash\libsnowcrash.vcxproj" (6) on node 1 (default targets).
     6>InitializeBuildStatus:
         Creating "Release\obj\libsnowcrash\libsnowcrash.unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libsnowcrash.vcxproj -> C:\Projects\drafter\build\Release\lib\libsnowcrash.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libsnowcrash\libsnowcrash.unsuccessfulbuild".
         Touching "Release\obj\libsnowcrash\libsnowcrash.lastbuildstate".
     6>Done Building Project "C:\Projects\drafter\build\ext\snowcrash\libsnowcrash.vcxproj" (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       ext\snowcrash\libmarkdownparser.vcxproj" (7) on node 1 (default targets).
     7>InitializeBuildStatus:
         Creating "Release\obj\libmarkdownparser\libmarkdownparser.unsuccessfulbuild" because "AlwaysCreate" was specif
         ied.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libmarkdownparser.vcxproj -> C:\Projects\drafter\build\Release\lib\libmarkdownparser.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libmarkdownparser\libmarkdownparser.unsuccessfulbuild".
         Touching "Release\obj\libmarkdownparser\libmarkdownparser.lastbuildstate".
     1>Project "C:\Projects\drafter\build\drafter.sln" (1) is building "C:\Projects\drafter\build\drafter.vcxproj.metap
       roj" (2) on node 1 (default targets).
     2>Project "C:\Projects\drafter\build\drafter.vcxproj.metaproj" (2) is building "C:\Projects\drafter\build\libdraft
       er.vcxproj" (5) on node 2 (default targets).
     5>InitializeBuildStatus:
         Touching "Release\obj\libdrafter\libdrafter.unsuccessfulbuild".
     7>Done Building Project "C:\Projects\drafter\build\ext\snowcrash\libmarkdownparser.vcxproj" (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       ext\snowcrash\libsundown.vcxproj" (8) on node 1 (default targets).
     8>InitializeBuildStatus:
         Creating "Release\obj\libsundown\libsundown.unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
       Lib:
         All outputs are up-to-date.
         libsundown.vcxproj -> C:\Projects\drafter\build\Release\lib\libsundown.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libsundown\libsundown.unsuccessfulbuild".
         Touching "Release\obj\libsundown\libsundown.lastbuildstate".
     8>Done Building Project "C:\Projects\drafter\build\ext\snowcrash\libsundown.vcxproj" (default targets).
     5>ClCompile:
         C:\Program Files\Microsoft Visual Studio 11.0\VC\bin\CL.exe /c /I"..\ext\snowcrash\ext\markdown-parser\src" /I
         "..\ext\snowcrash\ext\markdown-parser\ext\sundown\src" /I..\ext\sos\src /I..\ext\snowcrash\src /Zi /nologo /W1
          /WX- /Od /Ob2 /Oi /Ot /Oy /GL /D BUILDING_SNOWCRASH=1 /D WIN32 /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO
         _DEPRECATE /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /GR /Fo"Release\obj\libdrafter\\" /Fd"R
         elease\obj\libdrafter\vc110.pdb" /Gd /TP /analyze- /errorReport:queue /MP ..\src\cdrafter.cc ..\src\Serialize.
         cc ..\src\SerializeAST.cc ..\src\SerializeSourcemap.cc ..\src\SerializeResult.cc ..\src\RefractAPI.cc ..\src\R
         efractDataStructure.cc ..\src\RefractParseResult.cc ..\src\Render.cc
         cdrafter.cc
         Serialize.cc
         SerializeAST.cc
         SerializeSourcemap.cc
         SerializeResult.cc
         RefractAPI.cc
         RefractDataStructure.cc
         RefractParseResult.cc
     5>..\src\RefractDataStructure.cc(297): warning C4348: 'drafter::ExtractValueMember<T>::Store' : redefinition of de
       fault parameter : parameter 2 [C:\Projects\drafter\build\libdrafter.vcxproj]
                 with
                 [
                     T=ElementType
                 ]
                 ..\src\RefractDataStructure.cc(297) : see declaration of 'drafter::ExtractValueMember<T>::Store'
                 with
                 [
                     T=ElementType
                 ]
                 ..\src\RefractDataStructure.cc(512) : see reference to class template instantiation 'drafter::ExtractV
         alueMember<T>' being compiled
                 with
                 [
                     T=ElementType
                 ]
                 ..\src\RefractDataStructure.cc(641) : see reference to function template instantiation 'refract::IElem
         ent *drafter::RefractElementFromValue<T>(const mson::ValueMember &,const mson::BaseTypeName)' being compiled
                 with
                 [
                     T=refract::BooleanElement
                 ]
                 ..\src\RefractDataStructure.cc(650) : see reference to function template instantiation 'drafter::Value
         Trait::ElementType *drafter::ValueTrait::Invoke<refract::BooleanElement>(const drafter::ValueTrait::InputType
         &,const mson::BaseTypeName)' being compiled
                 ..\src\RefractDataStructure.cc(742) : see reference to function template instantiation 'refract::IElem
         ent *drafter::MsonMemberToRefract<drafter::ValueTrait>(const drafter::ValueTrait::InputType &,const mson::Base
         TypeName)' being compiled
                 ..\src\RefractDataStructure.cc(297) : see declaration of 'drafter::ExtractValueMember<T>::Store'
                 with
                 [
                     T=ElementType
                 ]
                 ..\src\RefractDataStructure.cc(297) : see declaration of 'drafter::ExtractValueMember<T>::Store'
                 with
                 [
                     T=ElementType
                 ]
                 ..\src\RefractDataStructure.cc(297) : see declaration of 'drafter::ExtractValueMember<T>::Store'
                 with
                 [
                     T=ElementType
                 ]
                 ..\src\RefractDataStructure.cc(297) : see declaration of 'drafter::ExtractValueMember<T>::Store'
                 with
                 [
                     T=ElementType
                 ]
         Render.cc
       Lib:
         C:\Program Files\Microsoft Visual Studio 11.0\VC\bin\Lib.exe /OUT:"C:\Projects\drafter\build\Release\lib\libdr
         after.lib" /NOLOGO /LTCG Release\obj\libdrafter\drafter.obj
         Release\obj\libdrafter\cdrafter.obj
         Release\obj\libdrafter\Serialize.obj
         Release\obj\libdrafter\SerializeAST.obj
         Release\obj\libdrafter\SerializeSourcemap.obj
         Release\obj\libdrafter\SerializeResult.obj
         Release\obj\libdrafter\RefractAPI.obj
         Release\obj\libdrafter\RefractDataStructure.obj
         Release\obj\libdrafter\RefractParseResult.obj
         Release\obj\libdrafter\Render.obj
         Release\obj\libdrafter\Element.obj
         Release\obj\libdrafter\SerializeCompactVisitor.obj
         Release\obj\libdrafter\SerializeVisitor.obj
         Release\obj\libdrafter\ComparableVisitor.obj
         Release\obj\libdrafter\TypeQueryVisitor.obj
         Release\obj\libdrafter\IsExpandableVisitor.obj
         Release\obj\libdrafter\ExpandVisitor.obj
         Release\obj\libdrafter\RenderJSONVisitor.obj
         Release\obj\libdrafter\Registry.obj
         libdrafter.vcxproj -> C:\Projects\drafter\build\Release\lib\libdrafter.lib
       FinalizeBuildStatus:
         Deleting file "Release\obj\libdrafter\libdrafter.unsuccessfulbuild".
         Touching "Release\obj\libdrafter\libdrafter.lastbuildstate".
     5>Done Building Project "C:\Projects\drafter\build\libdrafter.vcxproj" (default targets).
     3>Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (3) is building "C:\Projects\drafter\build\
       test-libdrafter.vcxproj" (9) on node 1 (default targets).
     9>InitializeBuildStatus:
         Creating "Release\obj\test-libdrafter\test-libdrafter.unsuccessfulbuild" because "AlwaysCreate" was specified.
     2>Project "C:\Projects\drafter\build\drafter.vcxproj.metaproj" (2) is building "C:\Projects\drafter\build\drafter.
       vcxproj" (10) on node 2 (default targets).
    10>InitializeBuildStatus:
         Creating "Release\obj\drafter\drafter.unsuccessfulbuild" because "AlwaysCreate" was specified.
     9>ClCompile:
         C:\Program Files\Microsoft Visual Studio 11.0\VC\bin\CL.exe /c /I..\src /I..\test /I..\test\vendor\Catch\inclu
         de /I..\ext\snowcrash\src /I"..\ext\snowcrash\ext\markdown-parser\src" /I"..\ext\snowcrash\ext\markdown-parser
         \ext\sundown\src" /I..\ext\sos\src /Zi /nologo /W1 /WX- /Od /Ob2 /Oi /Ot /Oy /GL /D BUILDING_SNOWCRASH=1 /D WI
         N32 /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D WIN /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wch
         ar_t /Zc:forScope /GR /Fo"Release\obj\test-libdrafter\\" /Fd"Release\obj\test-libdrafter\vc110.pdb" /Gd /TP /a
         nalyze- /errorReport:queue /MP "..\test\test-RefractAPITest.cc" "..\test\test-RefractDataStructureTest.cc" "..
         \test\test-RefractParseResultTest.cc" "..\test\test-RenderTest.cc" "..\test\test-SerializeResult.cc" "..\test\
         test-cdrafter.cc" "..\test\test-drafter.cc"
    10>ClCompile:
         C:\Program Files\Microsoft Visual Studio 11.0\VC\bin\CL.exe /c /I..\ext\cmdline /I"..\ext\snowcrash\ext\markdo
         wn-parser\src" /I"..\ext\snowcrash\ext\markdown-parser\ext\sundown\src" /I..\ext\sos\src /I..\ext\snowcrash\sr
         c /Zi /nologo /W1 /WX- /Od /Ob2 /Oi /Ot /Oy /GL /D BUILDING_SNOWCRASH=1 /D WIN32 /D _CRT_SECURE_NO_DEPRECATE /
         D _CRT_NONSTDC_NO_DEPRECATE /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /GR /Fo"Release\obj\dr
         after\\" /Fd"Release\obj\drafter\vc110.pdb" /Gd /TP /analyze- /errorReport:queue /MP ..\src\main.cc
     9>ClCompile:
         test-RefractAPITest.cc
         test-RefractDataStructureTest.cc
    10>ClCompile:
         main.cc
     9>ClCompile:
         test-RefractParseResultTest.cc
         test-RenderTest.cc
    10>Link:
         C:\Program Files\Microsoft Visual Studio 11.0\VC\bin\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Projects\drafter\bui
         ld\Release\drafter.exe" /INCREMENTAL:NO /NOLOGO /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /m
         anifest:embed /PDB:"C:\Projects\drafter\build\Release\drafter.pdb" /SUBSYSTEM:CONSOLE /OPT:REF /OPT:ICF /LTCG
         /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:\Projects\drafter\build\Release\drafter.lib" /MACHINE:X86 /SAFESEH
         Release\obj\drafter\main.obj
         Release\obj\drafter\config.obj
         Release\obj\drafter\reporting.obj
         C:\Projects\drafter\build\Release\lib\libdrafter.lib
         C:\Projects\drafter\build\Release\lib\libsos.lib
         C:\Projects\drafter\build\Release\lib\libsnowcrash.lib
         C:\Projects\drafter\build\Release\lib\libmarkdownparser.lib
         C:\Projects\drafter\build\Release\lib\libsundown.lib
         Generating code
     9>ClCompile:
         test-SerializeResult.cc
         test-cdrafter.cc
    10>Link:
         Finished generating code
         drafter.vcxproj -> C:\Projects\drafter\build\Release\\drafter.exe
       FinalizeBuildStatus:
         Deleting file "Release\obj\drafter\drafter.unsuccessfulbuild".
         Touching "Release\obj\drafter\drafter.lastbuildstate".
    10>Done Building Project "C:\Projects\drafter\build\drafter.vcxproj" (default targets).
     2>Done Building Project "C:\Projects\drafter\build\drafter.vcxproj.metaproj" (default targets).
     9>ClCompile:
         test-drafter.cc
       Link:
         C:\Program Files\Microsoft Visual Studio 11.0\VC\bin\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Projects\drafter\bui
         ld\Release\test-libdrafter.exe" /INCREMENTAL:NO /NOLOGO /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='fa
         lse'" /manifest:embed /PDB:"C:\Projects\drafter\build\Release\test-libdrafter.pdb" /SUBSYSTEM:CONSOLE /OPT:REF
          /OPT:ICF /LTCG /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:\Projects\drafter\build\Release\test-libdrafter.lib
         " /MACHINE:X86 /SAFESEH "Release\obj\test-libdrafter\test-drafter.obj"
         "Release\obj\test-libdrafter\test-cdrafter.obj"
         "Release\obj\test-libdrafter\test-SerializeResult.obj"
         "Release\obj\test-libdrafter\test-RefractDataStructureTest.obj"
         "Release\obj\test-libdrafter\test-RefractAPITest.obj"
         "Release\obj\test-libdrafter\test-RefractParseResultTest.obj"
         "Release\obj\test-libdrafter\test-RenderTest.obj"
         C:\Projects\drafter\build\Release\lib\libdrafter.lib
         C:\Projects\drafter\build\Release\lib\libsos.lib
         C:\Projects\drafter\build\Release\lib\libsnowcrash.lib
         C:\Projects\drafter\build\Release\lib\libmarkdownparser.lib
         C:\Projects\drafter\build\Release\lib\libsundown.lib
            Creating library C:\Projects\drafter\build\Release\test-libdrafter.lib and object C:\Projects\drafter\build
         \Release\test-libdrafter.exp
         Generating code
         Finished generating code
         test-libdrafter.vcxproj -> C:\Projects\drafter\build\Release\\test-libdrafter.exe
       FinalizeBuildStatus:
         Deleting file "Release\obj\test-libdrafter\test-libdrafter.unsuccessfulbuild".
         Touching "Release\obj\test-libdrafter\test-libdrafter.lastbuildstate".
     9>Done Building Project "C:\Projects\drafter\build\test-libdrafter.vcxproj" (default targets).
     3>Done Building Project "C:\Projects\drafter\build\test-libdrafter.vcxproj.metaproj" (default targets).
     1>Done Building Project "C:\Projects\drafter\build\drafter.sln" (default targets).

Build succeeded.

Time Elapsed 00:01:07.86
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,10 +1,15 @@
 version: 1.0.{build}
 
+environment:
+  matrix:
+#   - VS_VERSION: MSVC2012
+    - VS_VERSION: MSVC2013
+
 build_script:
   - git submodule update --init --recursive
   - gem install bundler
-  - vcbuild.bat MSVC2013
+  - vcbuild.bat %VS_VERSION%
 
 test_script:
-  - vcbuild.bat MSVC2013 test
-  - vcbuild.bat MSVC2013 inttest
+  - vcbuild.bat %VS_VERSION% test
+  - vcbuild.bat %VS_VERSION% inttest

--- a/src/SerializeAST.cc
+++ b/src/SerializeAST.cc
@@ -493,7 +493,7 @@ sos::Object WrapAsset(const Asset& asset, const AssetRole& role)
     return assetObject;
 }
 
-sos::Object WrapPayload(const Payload& payload, const Action* action = NULL)
+sos::Object WrapPayload(const Payload& payload, const Action* action)
 {
     sos::Object payloadObject;
 
@@ -663,7 +663,7 @@ sos::Object WrapResource(const Resource& resource)
     resourceObject.set(SerializeKey::URITemplate, sos::String(resource.uriTemplate));
 
     // Model
-    sos::Object model = (resource.model.name.empty() ? sos::Object() : WrapPayload(resource.model));
+    sos::Object model = (resource.model.name.empty() ? sos::Object() : WrapPayload(resource.model, NULL));
     resourceObject.set(SerializeKey::Model, model);
 
     // Parameters

